core/src/main/java/org/infinispan/interceptors/locking/OptimisticLockingInterceptor.java

```diff
@@ -139,6 +89,28 @@ public Object visitPutMapCommand(InvocationContext ctx, PutMapCommand command) t
    }
 
    @Override
+   public Object visitLockControlCommand(TxInvocationContext ctx, LockControlCommand command) throws Throwable {
+      throw new InvalidCacheUsageException("Explicit locking is not allowed with optimistic caches!");
+   }
+
+   @Override
+   protected Log getLog() {
+      return log;
+   }
```

Comment:
my beautiful IDEA orders the methods by visibility: public -> protected -> private :D 

I'll put the order back :P